apisix/plugins/limit-count.lua

```diff
@@ -36,11 +36,10 @@ local schema = {
     properties = {
         count = {type = "integer", exclusiveMinimum = 0},
         time_window = {type = "integer",  exclusiveMinimum = 0},
-        key = {
-            type = "string",
-            enum = {"remote_addr", "server_addr", "http_x_real_ip",
-                    "http_x_forwarded_for", "consumer_name", "service_id"},
-            default = "remote_addr",
+        key = {type = "string", default = "remote_addr"},
+        key_type = {type = "string",
+            enum = {"var", "var_combination"},
```

Comment:
It looks like the constraints for `var` losing.

Reply:
Thanks @tokers , if we keep the enum constraints for var like before, it will be a problem to use var_combination, because var_combination provides a more flexible way to key selecting.

Reply:
So it's my understanding that a potential problem will be introduced here, users might configure a non-existent variable and the validation is OK, but actually, the rate-limiting won't be the effect?

I think maybe some documents should be added so that at least we can give some hints to users.

Reply:
Thanks @tokers  I am working on it, update it later~

Reply:
OK, I saw the changes.

Reply:
please add test cases for non-existent variable @Xunzhuo